V2/src/fc_state_machine.cpp

```diff
@@ -2,57 +2,164 @@
 
 #include <fc_state_machine.h>
 #include <error_checker.h>
-
 #include <pin_defs.h>
+#include <analogs.h>
+
+#define CAP_THRESHOLD 20 // Move to another file, if they aren't already.
+
+void update_leds(void);
 
 #ifdef ENABLE_FC_TEST
 uint32_t state = FC_TEST;
 #else
-uint32_t state = FC_INIT;
+uint32_t state = FC_STANDBY;
 #endif
 
-uint32_t get_fc_state() {
-    // read-only access to state variable
+DigitalIn start(START); 
+
+// Valves.
+DigitalOut supply_v(SUPPLY_V);
+DigitalOut purge_v(PURGE_V);
+DigitalOut other1_v(VALVE3);
+DigitalOut other2_v(VALVE4);
+
+// Relays.
+DigitalOut start_r(START_R);
+DigitalOut motor_r(MOTOR_R);
+DigitalOut charge_r(CHARGE_R);
+DigitalOut cap_r(CAP_R);
+DigitalOut fcc_r(FCC_R);
+
+// LEDs.
+DigitalOut alarm_led(ALARM_LED);
+DigitalOut debug_led(DEBUG_LED);
+DigitalOut shut_led(SHUT_LED);
+DigitalOut run_led(RUN_LED);
+DigitalOut start_led(START_LED);
+
+uint32_t get_fc_state() // Read-only access to state variable.
+{
     return state;
 }
 
-// todo - other digital pin defs
-DigitalIn start(START);
+void fc_state_machine_thread() 
+{
+    fcc_r.write(true); // Solid state relay.
 
-// todo - define which timers we need
-Timer timer;
-
-void fc_state_machine_thread() {
-    while (true) {
-        if (check_all_errors()) {
-            state = FC_FAULT;
-        }
+    while (true) 
+    {
+        if (check_all_errors()) 
+            state = FC_ALARM;
 
-        if (state == FC_TEST) {
+        if (FC_TEST == state) 
+        {
             ThisThread::sleep_for(50); // Add test state code here, if desired.
+            charge_r.write(false);
         }
 
-        if (state == FC_INIT){
-            if (start) {
-                state = FC_STARTUP;
-                // we will need lots if timers
-                // timer example:
-                timer.start();
+        if (FC_STANDBY == state) // Wait until start button is pressed.
+        { 
+            if (start) 
+            {
+                purge_v.write(false); // Perform start purge.
+                start_r.write(true);
+                motor_r.write(false);
+                charge_r.write(false);
+                cap_r.write(false);
+                supply_v.write(true);
+                Thread::wait(1000);
+                purge_v.write(true);
+                Thread::wait(200);
+                purge_v.write(false); 
+
+                supply_v.write(true); // Charge state setup.
+                start_r.write(false);
+                motor_r.write(false);
+                charge_r.write(true);
```

Comment:
I was thinking to avoid the startup/charge issue we could just put if statements around when either of those are changed that say 
if (startup relay is open) { close charge relay}
else {error}
Because I don't want to depend on an error checker detecting it after it happens.

Reply:
Issue is we arent really setup for setting errors from the statemachine. Requires a bit of redesign

Reply:
Could we just set the state to be FC_FAULT?